favor/favor.py

```diff
@@ -15,6 +13,8 @@ def __init__(
         self,
         key_dim,
         orthonormal=True,
+        multihead=True,
```

Comment:
I don't really like including the multihead in the base class, because it makes the thing less readable IMO

could you create a derived MultiHeadFAVOR, that create several FAVOR ? This would not be equivalent ?

Reply:
This is just for convenience, equivalently you can just reshape before and after the attention. So yes, maybe we can instead have a wrapper which will do that.

Another option would be to just allow any number of dimensions before *and* after the actual feature dimension (and just flatten them as you already do). So for single-head attention we would have `dim=1`, for multi-head `dim=2`; if for some reason you didn't have batch dimension, you could have `dim=0`...

Reply:
thing is: IMO, multihead attention is indeed just a matter ok the transforms to obtain the keys, values, queries, right ? The actual attention layer is the same in all cases. I think that flattening is indeed the way to go

Reply:
OK, reverting the multi-head thing.